googletest/CMakeLists.txt

```diff
@@ -0,0 +1,7 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+add_subdirectory(googletest)
+
+project(DirectX-Headers-GoogleTest-Suite CXX)
+add_executable(Feature-Support-Test feature_support_test.cpp)
+target_link_libraries(Feature-Support-Test DirectX-Headers DirectX-Guids d3d12 dxcore gtest_main)
```

Comment:
Google tests works fine with CMake, but when it targets Visual Studio I have to manually change the code generation method from `MDd` to `MTd` in order for it to compile. I checked online and it can be fixed by adding a compile flag to the CMakeFile. Should I add this flag to our project?

Reply:
Seems like a good idea.

Reply:
Added the compiler flags in `googletest/CMakeFiles.txt`. The workaround comes from [CMake community wiki](https://gitlab.kitware.com/cmake/community/-/wikis/FAQ#how-can-i-build-my-msvc-application-with-a-static-runtime). Not the most elegant solution, but should eliminate the need to tweak the settings manually in Visual Studio.